components/PlantList.tsx

```diff
@@ -1,24 +1,172 @@
+    ['JUNE', 5],
+    ['JULY', 6],
+    ['AUGUST', 7],
+    ['SEPTEMBER', 8],
+    ['OCTOBER', 9],
+    ['NOVEMBER', 10],
+    ['DECEMBER', 11],
+  ]);
 
   useEffect(() => {
     const fetchPlantSeasonality = async () => {
-      const plantList = await getPlantSeasonality('Tennessee');
-      setPlants(plantList);
+      // gets plants in Tennessee by default
+      const plantList = await getAllPlants();
+      const us_state = 'TENNESSEE';
+      const filteredPlantList = plantList.filter(
+        plant => plant.us_state === us_state,
+      );
+      setPlants(filteredPlantList);
     };
 
     fetchPlantSeasonality();
   }, []);
 
+  // Check if growingSeason matches the plant's growing season
+  const checkGrowingSeason = (plant: Plant) => {
+    // Automatically returns true if selected growing season is ''
+    if (!growingSeason) {
+      return true;
+    }
+
+    // list of valid indexes for the growing season
+    // indexes are the months of the year
+    const validIndexes = growingSeasonToIndex.get(growingSeason);
+
+    const isInRange = (start: number, end: number, validIndexes: number[]) => {
+      // Checks if the start and end months are within the valid range
+      if (start <= end) {
+        return validIndexes.some(index => index >= start && index <= end);
+      } else {
+        // Handle wrap-around case (e.g. NOVEMBER to FEBRUARY)
+        return validIndexes.some(index => index >= start || index <= end);
+      }
+    };
+
+    // Handle late/early month logic -- Set late/early month to just the month
+    let indoorsStart = plant.indoors_start;
+    let indoorsEnd = plant.indoors_end;
+    let outdoorsStart = plant.outdoors_start;
+    let outdoorsEnd = plant.outdoors_end;
+
+    /// If field is not null and starts with 'LATE' or 'EARLY,
+    // get substring after 'LATE_ or 'EARLY_'
+    if (
+      indoorsStart &&
+      (indoorsStart.startsWith('LATE') || indoorsStart.startsWith('EARLY'))
+    ) {
+      indoorsStart = indoorsStart.substring(5);
+    }
+    if (
+      indoorsEnd &&
+      (indoorsEnd.startsWith('LATE') || indoorsEnd.startsWith('EARLY'))
+    ) {
+      indoorsEnd = indoorsEnd.substring(5);
+    }
+    if (
+      outdoorsStart &&
+      (outdoorsStart.startsWith('LATE') || outdoorsStart.startsWith('EARLY'))
+    ) {
+      outdoorsStart = outdoorsStart.substring(5);
+    }
+    if (
+      outdoorsEnd &&
+      (outdoorsEnd.startsWith('LATE') || outdoorsEnd.startsWith('EARLY'))
+    ) {
+      outdoorsEnd = outdoorsEnd.substring(5);
+    }
+
+    // Checks if either indoor_start to indoor_end or outdoor_start to outdoor_end
+    // is within the valid range
+    // exclamation marks to assert values are not undefined
+    return (
+      isInRange(
+        monthToIndex.get(indoorsStart)!,
+        monthToIndex.get(indoorsEnd)!,
+        validIndexes!,
+      ) ||
+      isInRange(
+        monthToIndex.get(outdoorsStart)!,
+        monthToIndex.get(outdoorsEnd)!,
+        validIndexes!,
```

Comment:
great handle! i was originally thinking that we'd only check 
```
isInRange(
        monthToIndex.get(indoorsStart)!,
        monthToIndex.get(outdoorsEnd)!,
        validIndexes!)
```
However, this logic seems more accurate. i'll check w POC, but this is great for now